plugin_tests/submission_test.py

```diff
@@ -0,0 +1,395 @@
+
+from tests import base
+
+
+def setUpModule():
+    base.enabledPlugins.append('covalic')
+    base.startServer()
+
+
+def tearDownModule():
+    base.stopServer()
+
+
+class SubmissionBase(base.TestCase):
+    def setUp(self):
+        super(SubmissionBase, self).setUp()
+
+        from girder.plugins.jobs.models.job import Job
+        from girder.plugins.worker.constants import PluginSettings as WorkerSettings
+        from girder.plugins.covalic.constants import PluginSettings as CovalicSettings
+
+        self.scheduleJobMock = mock.patch.object(Job, 'scheduleJob')
+        self.scheduleJobMock.start()
+
+        self.model('setting').set(WorkerSettings.API_URL, 'http://example.com/api/v1')
+
+        user = {
+            'email': 'admin@email.com',
+            'login': 'amdinlogin',
+            'firstName': 'First',
+            'lastName': 'Last',
+            'password': 'adminpassword',
+            'admin': True
+        }
+        self.admin = self.model('user').createUser(**user)
+
+        user = {
+            'email': 'good@email.com',
+            'login': 'goodlogin',
+            'firstName': 'First',
+            'lastName': 'Last',
+            'password': 'goodpassword',
+            'admin': False
+        }
+        self.user = self.model('user').createUser(**user)
+
+        self.group = self.model('group').createGroup('phase1', self.admin)
+        self.model('group').addUser(self.group, self.user)
+
+        user = {
+            'email': 'scoring@email.com',
+            'login': 'scoringlogin',
+            'firstName': 'First',
+            'lastName': 'Last',
+            'password': 'scoringpassword',
+            'admin': False
+        }
+        self.scoringUser = self.model('user').createUser(**user)
+        self.model('setting').set(
+            CovalicSettings.SCORING_USER_ID,
+            self.scoringUser['_id']
+        )
+
+        self.challenge = self.model('challenge', 'covalic').createChallenge(
+            name='challenge 1',
+            creator=self.admin,
+            public=False)
+
+        metrics = {
+            'accuracy': {'weight': 1}
+        }
+        self.phase1 = self.model('phase', 'covalic').createPhase(
+            'phase 1', self.challenge, creator=self.admin, ordinal=1
+        )
+        self.phase1['metrics'] = metrics
+        self.phase1['active'] = True
+        self.phase1['participantGroupId'] = self.group['_id']
+        self.model('phase', 'covalic').save(self.phase1)
+        self.phase2 = self.model('phase', 'covalic').createPhase(
+            'phase 2', self.challenge, creator=self.admin, ordinal=2
+        )
+        self.phase2['metrics'] = metrics
+        self.model('phase', 'covalic').save(self.phase2)
+
+    def tearDown(self):
+        self.scheduleJobMock.stop()
+        super(SubmissionBase, self).tearDown()
+
+    def createSubmission(self, phase, user, title, score=None, **kwargs):
+        scoreDict = [{
+            'dataset': 'Average',
```

Comment:
Validating/saving a submission computes the 'Average' dataset, so I don't think it should be added explicitly even in these tests. Instead, supply two datasets with distinct scores.